platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -0,0 +1,626 @@
+                                                pSignature->size ) ) /*lint !e732 !e9034 Allow comparison in this context. */
+                {
+                    LogError( ( "File signature check failed at FINAL" ) );
+                    mainErr = OtaPalSignatureCheckFailed;
+                }
+            }
+
+            /* Free the temporary file page buffer. */
+            OPENSSL_free( pBuf );
+        }
+        else
+        {
+            LogError( ( "Failed to allocate buffer memory." ) );
+            mainErr = OtaPalOutOfMemory;
+        }
+    }
+    else
+    {
+        LogError( ( "File signature check failed at INIT." ) );
+        mainErr = OtaPalSignatureCheckFailed;
+    }
+
+    return mainErr;
+}
+
+/* Verify the signature of the specified file using OpenSSL. */
+static OtaPalStatus_t otaPal_CheckFileSignature( OtaFileContext_t * const C )
+{
+    OtaPalMainStatus_t mainErr = OtaPalSuccess;
+    EVP_PKEY * pPkey = NULL;
+    EVP_MD_CTX * pSigContext = NULL;
+
+    if( C != NULL )
+    {
+        /* Extract the signer cert from the file */
+        pPkey = Openssl_GetPkeyFromCertificate( C->pCertFilepath );
+
+        /* Create a new signature context for verification purpose */
+        pSigContext = EVP_MD_CTX_new();
+
+        if( ( pPkey != NULL ) && ( pSigContext != NULL ) )
+        {
+            /* Verify an ECDSA-SHA256 signature. */
+            mainErr = Openssl_DigestVerify( pSigContext, pPkey, C->pFile, C->pSignature );
+        }
+        else
+        {
+            if( pSigContext == NULL )
+            {
+                LogError( ( "File signature check failed at NEW sig context." ) );
+                mainErr = OtaPalSignatureCheckFailed;
+            }
+            else
+            {
+                LogError( ( "File signature check failed at EXTRACT pkey from signer certificate." ) );
+                mainErr = OtaPalBadSignerCert;
+            }
+        }
+
+        /* Free up objects */
+        EVP_MD_CTX_free( pSigContext );
+        EVP_PKEY_free( pPkey );
+    }
+    else
+    {
+        /* FIXME: Invalid error code for a NULL file context. */
+        LogError( ( "Failed to check file signature: Paramater check failed: "
+                    " Invalid OTA file context." ) );
+        /* Invalid OTA context or file pointer. */
+        mainErr = OtaPalNullFileContext;
+    }
+
+    return OTA_PAL_COMBINE_ERR( mainErr, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+OtaPalStatus_t otaPal_ResetDevice( OtaFileContext_t * const C )
+{
+    ( void ) C;
+
+    /* Return no error.  linux implementation does not reset device. */
+    return OTA_PAL_COMBINE_ERR( OtaPalSuccess, 0 );
+}
+
+/*-----------------------------------------------------------*/
+
+OtaPalStatus_t otaPal_ActivateNewImage( OtaFileContext_t * const C )
+{
+    ( void ) C;
+
+    /* Return no error. linux implementation simply does nothing on activate.
```

Comment:
`POSIX != Linux`.  Also, executables on POSIX systems usually aren't suffixed with `.exe` so I'd just delete line 482.  In addition, this PAL seems to just implement a file download workflow so it wouldn't be limited to executables anyway.